src/couch_replicator/test/eunit/couch_replicator_doc_ids_tests.erl

```diff
@@ -0,0 +1,83 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_replicator_doc_ids_tests).
+
+-include_lib("couch/include/couch_eunit.hrl").
+-include_lib("couch/include/couch_db.hrl").
+
+-define(DOC1, <<"doc1">>).
+-define(DOC2, <<"doc2">>).
+-define(DDOC3, <<"_design/doc3">>).
+
+doc_ids_replication_test_() ->
+    {
+        "Doc IDs filtered replication tests",
+        {
+            foreach,
+            fun couch_replicator_test_helper:test_setup/0,
+            fun couch_replicator_test_helper:test_teardown/1,
+            [
+                ?TDEF_FE(t_should_replicate_with_ddoc_ids)
+            ]
+        }
+    }.
+
+t_should_replicate_with_ddoc_ids({_Ctx, {Source, Target}}) ->
+    create_docs(Source),
+    replicate(#{
+        <<"source">> => db_url(Source),
+        <<"target">> => db_url(Target),
+        <<"doc_ids">> => [?DOC1, ?DDOC3]
```

Comment:
I guess you could factor out `DocIds = [?DOC1, ?DDOC3]` and then change the filter function to `lists:member(DocId, DocIds)` to DRY it out a little, but probably not worth it.

Reply:
Oh neat. Good idea. Or even have a `?DOC_IDS` macro as [?DOC1, ?DDOC3]`